examples/hello_piston.rs

```diff
@@ -1,24 +1,13 @@
 extern crate piston_window;
-extern crate glutin_window;
-extern crate piston;
-extern crate graphics;
 
-use std::cell::RefCell;
-use std::rc::Rc;
-use glutin_window::{ OpenGL, GlutinWindow };
 use piston_window::*;
-use piston::window::WindowSettings;
-
-use piston::event::*;
-use graphics::*;
-use piston::input::*;
 
 fn main() {
-    let window = Rc::new(RefCell::new(GlutinWindow::new(
-        OpenGL::_3_2,
-        WindowSettings::new("Hello Piston!", [640, 480])
+    let window = window(
+            OpenGL::_3_2,
```

Comment:
Hmmm, is there any chance the OpenGL version could be passed via a builder method to the WindowSettings? I'm imagining window construction becoming something like:

```Rust
let window = WindowSettings::new("Hello Piston!", [640, 480])
    .opengl(OpenGL::_3_2)
    .exit_on_esc(true)
    .build();
```

Is it possible to detect some default system opengl version if one isn't given?